geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/utils/GeoMessageSerializer.scala

```diff
@@ -95,9 +103,12 @@ object GeoMessageSerializer {
   */
 class GeoMessageSerializer(sft: SimpleFeatureType, lazyDeserialization: Boolean = false) extends LazyLogging {
 
-  import GeoMessageSerializer._
-
   private val serializer = {
+    val builder = KryoFeatureSerializer.builder(sft).withoutId.withUserData.immutable
+    if (lazyDeserialization) { builder.`lazy`.build() } else { builder.build() }
```

Comment:
Does it make sense to roll the lazy flag into the builder?

Reply:
hmm? it is part of the builder, it just seemed more concise to add the build inside the if, instead of assigning a new variable and then calling build on that. Or am I misunderstanding?

Reply:
Sorry, I can't read...  

That said, there's a tiny convenience method that could be added to the builder.  I'm thinking something like isLazy(boolean).  That said, this make sense now.